fix: correctly report the coverage of changed files when using the workingDirectory

paulswartz opened the pull request:
compare:
- (bad) https://github.com/mbta/rtr/pull/979#issuecomment-1064454846
- (good) https://github.com/paulswartz/data_platform/pull/1#issuecomment-1064571961

Note that the (good) comment includes the coverage of the files.

dependabot[bot] committed:
Bump lodash from 4.17.20 to 4.17.21 (#7)

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.20 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.20...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

zgosalvez committed:
npm audit fix

jbrownsw committed:
Uses -y to prevent the runner from hanging if lcov isn't installed (#8)

PoOwAa committed:
Branch-coverage (#12)

* Branch-coverage

* Standardize branch coverage for HTML reports

Co-authored-by: Zennon Gosalvez <1798166+zgosalvez@users.noreply.github.com>

zgosalvez committed:
Updates dependencies

paulswartz committed:
fix: correctly report changed files when using workingDirectory

ryan-mahoney commented on dist/main/index.js:
```diff
@@ -14536,8 +14537,15 @@ function ownerRepo(url) {
+async function genhtml(coverageFiles, tmpPath, workingDirectory) {
   const artifactName = core.getInput('artifact-name').trim();
   const artifactPath = path.resolve(tmpPath, 'html').trim();
   const args = [...coverageFiles];
```
minor js style, `concat` allows this to be one line, with no mutations:

```
const args = [...coverageFiles].concat(['--output-directory', artifactPath])
```

paulswartz replied:
Turns you you can also do `[...coverageFiles, '--output-directory', artifactPath]`!

ryan-mahoney replied:
of course -- even better!

ryan-mahoney replied:
You are probably aware, but JS has this syntax for objects which is similar to Elixir, but I haven't noticed in our code too often in the past:

```
const someObj  = {...anotherObj, abc: "def", [someVar]: "qrs"};
```

ryan-mahoney commented on dist/main/index.js:
```diff
@@ -14536,8 +14537,15 @@ function ownerRepo(url) {
+  return path + "/";
+}
+
+async function genhtml(coverageFiles, tmpPath, workingDirectory) {
```
minor: some linters would ding that this function was used before it was defined, ie, out of order

ryan-mahoney reviewed (approved):
Looks great, left some minor comments related to style

paulswartz committed:
Merge remote-tracking branch 'upstream/main' into ps-working-directory

paulswartz committed:
fix: cleanup lint/review errors

- out-of-order functions
- extra semicolons
- extra escaping
- mutation of lists
